### DISCUSSION TITLE
Can I see the content of registers?

### DISCUSSION THREAD

**Discussion by micdonato** [vscode-neovim/vscode-neovim#2276] [Q&A] [answered]
Probably a silly question but I am used to type `:registers` in neovim just to see what is in there.
Granted, all the other register functionalities work (I don't do much but I do yank to registers to paste over other stuff), but I was wondering if there was a way to actually see the content of all the registers in the extension.

**Comment by justinmk** [comment, marked as answer]
`:registers` should work, did you try it? Nvim output is sent to the `vscode-neovim messages` Output view (which should be auto-focused when you run a command that has multiline output).

![image](https://github.com/user-attachments/assets/be4e4cc1-620a-4044-ae93-17f3d1f6e610)

**Comment by micdonato** [comment]
Ohhhh I didn't see them there. The only thing I saw was this small event in the status bar. Thanks!
<img width="374" alt="Screenshot 2024-10-09 at 9 03 52 AM" src="https://github.com/user-attachments/assets/9d4be5c0-7c2d-4213-ac7a-c33a8a7907fb">
On the other hand, the output view does not focus automatically, but I think I can find out why

**Reply by micdonato** [reply]
That said, if I start a search and replace command, I see the output right away under the command, is there a chance to get registers there too? 
I attached an example with "chicken" and "lines"

<img width="638" alt="Screenshot 2024-10-10 at 1 20 32 PM" src="https://github.com/user-attachments/assets/2c0642aa-2e2f-4317-89f1-935e7e668db9">

**Reply by justinmk** [reply]
> is there a chance to get registers there too?

That's an interesting idea. I think `:s` with 'inccommand' is special-cased. cc @theol0403 

But I don't think showing output in the quickpick menu (which is what you are seeing) is a great UI, it doesn't allow interacting with the text. vscode Output channels exist for that reason.

**Reply by micdonato** [reply]
I see your point, although if I (anecdotal) read the output of `:registers` I am unlikely to interact with the text in the output: it is more likely that use it as reference and just use a particular register for `p`. 
At the end of the day what I would like (and believe me, I would love to be able to contribute, but I would not know where to start) is to have a quick reminder of what is in the registers.
For now I just moved the output to the sidebar and it honestly does the trick!

**Reply by justinmk** [reply]
If the Output isn't auto-focusing , try setting cmdheight=1 or cmdheight=2 to ensure that small outputs don't just use the statusbar.
